Octokit/Models/Response/BranchProtection.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Octokit
+{
+    /// <summary>
+    /// Protection details for a <see cref="Branch"/>.
+    /// Note: this is a PREVIEW api: https://developer.github.com/changes/2015-11-11-protected-branches-api/
+    /// </summary>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class BranchProtection
+    {
+        /// <summary>
+        /// Should this branch be protected or not
+        /// </summary>
+        public bool Enabled { get; set; }
+
+        /// <summary>
+        /// The <see cref="RequiredStatusChecks"/> information for this <see cref="Branch"/>.
+        /// </summary>
+        public RequiredStatusChecks RequiredStatusChecks { get; private set; }
+
+        public BranchProtection()
+        {
+            RequiredStatusChecks = new RequiredStatusChecks();
+        }
+
+        internal string DebuggerDisplay
+        {
+            get
+            {
+                return String.Format(CultureInfo.InvariantCulture, "Enabled: {0}", Enabled);
+            }
+        }
+    }
+
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class RequiredStatusChecks
+    {
+        /// <summary>
+        /// Who required status checks apply to
+        /// </summary>
+        public EnforcementLevel EnforcementLevel { get; set; }
+
+        /// <summary>
+        /// The list of status checks to require in order to merge into this <see cref="Branch"/>
+        /// </summary>
+        public ICollection<string> Contexts { get; private set; }
```

Comment:
FYI this one is implemented similar to the IssueUpdate object with it's list of Labels.  Implemented as a List<string> with public get and private set, then AddLabel() and ClearLabels() functions that manipulate the local private list.

I guess the difference is that IssueUpdate is a Request model object, whilst in this case the RequiredStatusChecks are in the Response section?  I can make it protected rather than private I guess